agency/detail/tuple_impl.hpp

```diff
@@ -160,13 +160,13 @@ template<class T, bool = tuple_use_empty_base_class_optimization<T>::value>
 class tuple_leaf_base
 {
   public:
-#if defined(__CUDACC__)
+#if defined(__CUDACC__) && !(defined(__clang__) && defined(__CUDA__))
```

Comment:
This looks correct, but I think rather than duplicating this line we should introduce a shorthand preprocessor symbol.

Since `optional.hpp` and `tuple_impl.hpp` are dependencies from external repositories, I will need to change them there and then pull them into Agency.